workshop/content/intermediate/templates/dynamic-references/_index.md

```diff
@@ -0,0 +1,238 @@
+
+In this lab, you will use AWS Secrets Manager to store database connection information such as *hostname*, *port*, *username*, and *password*. Next, you will use dynamic references to consume connection information from an `AWS::Lambda::Function` resource you describe in a template.
+
+Let’s get started! Choose to follow steps shown next:
+
+1. First, you will create an Amazon RDS database, and store connection information as a secret in AWS Secrets Manager:
+    1. Make sure you are in the `code/workspace/dynamic-references` directory.
+    2. Open the `database.yaml` CloudFormation template in your favorite text editor.
+    3. Note the following resources in the template:
+        1. the resource of type `AWS::RDS::DBInstance`, with which you describe your Amazon RDS instance
+        2. the resource of type `AWS::SecretsManager::Secret`, where you store database connection parameters as JSON key-value pairs in a secret named `DatabaseConnParams`:
+
+```json
+{
+    "RDS_HOSTNAME": "${Database.Endpoint.Address}",
+    "RDS_PORT": "${Database.Endpoint.Port}",
+    "RDS_USERNAME": "${DBUsername}",
+    "RDS_PASSWORD": "${DBPassword}"
+}
+```
+
+2. To Deploy the Database stack, follow the steps below:
+    1. Navigate to the [AWS CloudFormation Console](https://console.aws.amazon.com/cloudformation/), and choose **Create stack With new resources (standard)**.
+    2. In **Prepare template**, select **Template is ready**.
+    3. In **Template source**, select **Upload a template file**.
+    4. Choose the file `database.yaml`.
+    5. Enter a Stack name. For example, *cfn-workshop-database-stack.*
+    6. For `DBUsername`, specify the primary user name for the DB instance.
+    7. For `DBPassword`, specify the password for the primary user.
+    8. Choose to use default values for **Configure stack options**, and choose **Next**.
+    9. On the **Review** page for your stack, scroll down to the bottom, and choose **Create stack**.
+    10. You can view the progress of stack being created in the CloudFormation Console, by refreshing the stack creation page.
+    11. Refresh the page until you see the `CREATE_COMPLETE` status for your stack.
+
+3. Next, you will create an AWS Lambda Function, and pass database connection parameters as [Environment Variables](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html) to a Lambda function, by using dynamic references to the AWS Secrets Manager secret created earlier.
+    1. Make sure you are in the `code/workspace/dynamic-references` directory.
+    2. Open the `lambda_function.yaml` CloudFormation template in your favorite text editor.
+    3. The template describes an `AWS::Lambda::Function` resource type; update the template by appending the `Properties` section with the `Environment` property, with variables using dynamic references to the AWS Secret Manager secret you created earlier:
+
+```yaml
```

Comment:
- Please remove `!Sub` occurrences, as those are not needed here.
- Please remove credential references from environment variables (lines starting with `RDS_USERNAME` and `RDS_PASSWORD`) as recommended to increase database security (reference: https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html).